### PR TITLE
Update omnibus-overrides, switch to ruby 2.7

### PR DESCRIPTION
**Pull request by clintoncwolfe**
## Description

Switches from ruby 2.6.5 to 2.7.1 and updates bundler and rubygems to match Chef Infra Client.

## Related Issue

Part of #1059

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: omnibus_overrides.rb [hunk 1]
```diff
@@ -9,8 +9,8 @@
 override "chef-workstation-app", version: "v0.1.77"
 # /DO NOT MODIFY
 
-override :rubygems, version: "3.0.3" # rubygems ships its own bundler which may differ from bundler defined below and then we get double bundler which results in performance issues / CLI warnings. Make sure these versions match before bumping either.
-override :bundler, version: "1.17.2" # currently pinned to what ships in Ruby to prevent double bundler
+override :rubygems, version: "3.1.2" # rubygems ships its own bundler which may differ from bundler defined below and then we get double bundler which results in performance issues / CLI warnings. Make sure these versions match before bumping either.
+override :bundler, version: "2.1.4" # currently pinned to what ships in Ruby to prevent double bundler
```
Comment (tyler-ball) [inline omnibus_overrides.rb:13]: Hmmm, this one we may need to discuss. I know there was an issue with updating to > 2.x and I'm not sure if that issue still exists.

Comment (tyler-ball) [reply omnibus_overrides.rb:13]: This is what Chef is currently doing so we do that too

File: components/gems/Gemfile.lock [hunk 2]
```diff
@@ -394,6 +394,9 @@ GEM
     ffi (1.12.2)
     ffi (1.12.2-x64-mingw32)
     ffi (1.12.2-x86-mingw32)
+    ffi-compiler (1.0.1)
```
Comment (tyler-ball) [inline components/gems/Gemfile.lock:397]: @tas50 @lamont-granquist Is this going to bite me in the ass?

Comment (tas50) [reply components/gems/Gemfile.lock:397]: 🤷

Comment (tyler-ball) [reply components/gems/Gemfile.lock:397]: The new library which builds native extensions is https://github.com/cotag/http-parser which pulls in https://github.com/nodejs/http-parser/tree/77310eeb839c4251c07184a5db8885a572a08352 which is a C extension so... hopefully not?

tenor-129399655

Comment (tas50) [reply components/gems/Gemfile.lock:397]: Turns out the answer is YES

File: omnibus/config/software/more-ruby-cleanup.rb [hunk 3]
```diff
@@ -48,7 +47,6 @@
       ed25519.png
       example
       examples
-      ext
```
Comment (tas50) [inline omnibus/config/software/more-ruby-cleanup.rb:51]: Was this for nokogiri or the http gem mess. This is going to bloat the builds pretty significantly.

Comment (tyler-ball) [reply omnibus/config/software/more-ruby-cleanup.rb:51]: This was for the http gem - specifically without this it is missing `http-parser-ext`.

**Pull request merged**